runhttpd.sh

```diff
@@ -2,12 +2,30 @@
 
 PROVISIONING_INTERFACE=${PROVISIONING_INTERFACE:-"provisioning"}
 HTTP_PORT=${HTTP_PORT:-"80"}
-HTTP_IP=$(ip -4 address show dev "$PROVISIONING_INTERFACE" | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | head -n 1)
```

Comment:
nit - we should probably initialize this as it's going to be undefined the first time around the `until`

Reply:
What's the risk of not initializing it? It seemed unnecessary, as the check in `until` is "not empty", so until does loop correctly. Uninitialized var is empty.

Reply:
I guess I was just thinking of best-practice and future bug avoidance, if anyone decides to add set -eu (which would be a good idea IMO) in future this will break, and if anyone forgets the double quotes in the test it's liable to break with `[` vs `[[`

Not a blocker but it's a change from a previously defined variable to one that's undefined so personally I'd just define it before the loop.

Reply:
Initializing it to "" would be sufficient to avoid those cases? The problem was the `ip` command is slightly different depending on whether we're doing it based on waiting for an IP vs. waiting for an interface.

Reply:
Yeah I was just thinking we'd initialize with an empty string so we'd explicitly fail the test the first time around the loop.

Reply:
Fixed in ironic-common.sh